linuxkm/lkcapi_sha_glue.c

```diff
@@ -944,24 +971,34 @@ static int wc_linuxkm_drbg_generate(struct crypto_rng *tfm,
 {
     struct wc_linuxkm_drbg_ctx *ctx = (struct wc_linuxkm_drbg_ctx *)crypto_rng_ctx(tfm);
     int ret;
-
-    wc_LockMutex(&ctx->lock);
+    int my_cpu =
+        raw_smp_processor_id(); /* Note, core is not locked, so the actual core
```

Comment:
So your saying that multiple calls can use the same random indexed on the processor id?

Reply:
Prefer the comment above the variable declaration.

Reply:
>So your saying that multiple calls can use the same random indexed on the processor id?

Not sure what you're asking.  The core of the caller is random, so the exact DRBG it gets is random.  It may or may not see the same DRBG on sequential calls via the kernel glue layer.  All as expected+designed.

Reply:
Note that I don't have have the literal or figurative bandwidth to babysit another go through Jenkins just to bikeshed the comment formatting.  I'm on vacation.

Reply:
>The issue is whether you need to have a Mutex when you are indexing by process id.

Ah!  That's where it gets interesting.  I'm not disabling interrupts/preemption (yet), so the core can change _while executing the code_.  The mutex assures that is benign, resulting only in rare mutex contention.

The other reason the mutex is still needed is to allow the explicit user-requested reseeding operation., which acts on _all_ the DRBGs.

This is just stage 1 of the refactor, done ahead of schedule to unblock performance for a particular customer, but in the next stage I will indeed attempt to refactor out the mutex entirely, through core locking and lock-free trickery for the reseeds.